File: 《算法与竞赛操作软件对接及说明文档》/灵巧手接口文档/fourierdexhandsdk-main/example/python/fourier_dexhand_sdk_demo.py
import time
import dexhandpy.fdexhand as fdh

class DexhandDemo:
    def __init__(self):
        self.dh = fdh.DexHand()
        self.result = fdh.Ret

    def init(self):
        ret = self.dh.init()
        if (ret == self.result.SUCCESS):
            print("init successfully")
        else:
            print("init failed")
            exit(0)

    def reboot(self):
        ret = self.dh.reboot()
        if ret == self.result.SUCCESS:
            print("reboot successfully")
        else:
            print("reboot failed")

    def device_info(self):
        ip_list = self.dh.get_ip_list()
        for ip in ip_list:
            name = self.dh.get_name(ip)
            type = self.dh.get_type(ip)
            dri_ver = self.dh.get_driver_ver(ip)
            hw_ver = self.dh.get_hardware_ver(ip)
            print(f"{ip}, name:{name}, type:{type}, driver_ver:{dri_ver}, hardware_ver:{hw_ver}")

    def get_pos(self):
        ip_list = self.dh.get_ip_list()
        for ip in ip_list:
            pos = self.dh.get_pos(ip)
            print(f"{ip}, pos: {[f'{x:.2f}' for x in pos]}")

    def set_pos(self):
        ip_list = self.dh.get_ip_list()
        for ip in ip_list:
            if self.dh.get_name(ip) == "fdhv2":
                self.dh.set_pos(ip, [0] * 12)
            else:
                self.dh.set_pos(ip, [1] * 6)

    def get_ts_matrix(self):
        ip_list = self.dh.get_ip_list()
        for ip in ip_list:
            if self.dh.get_name(ip) == "fdhv2":
                print(f"{ip}, matrix: ")
                ts_matrix = self.dh.get_ts_matrix(ip)
                for row in ts_matrix:
                    print(f"{row}")

    def get_hand_config(self):
        ip_list = self.dh.get_ip_list()
        for ip in ip_list:
            config = self.dh.get_hand_config(ip)
            print(f"result {config.result}, ip: {config.ip}, gateway: {config.gateway}, sn: {config.sn}, mac: {config.mac}")

    def get_pvc(self):
        ip_list = self.dh.get_ip_list()
        for ip in ip_list:
            pvc = self.dh.get_pvc(ip)
            if pvc:
                print(f"{ip}, p: {[f'{x:.2f}' for x in pvc[0]]}")
                print(f"{ip}, v: {[f'{x:.2f}' for x in pvc[1]]}")
                print(f"{ip}, c: {[f'{x:.2f}' for x in pvc[2]]}")

    def set_pvc(self):
        ip_list = self.dh.get_ip_list()
        for ip in ip_list:
            pvc = []
            if (self.dh.get_name(ip) == "fdhv2"):
                pvc = [[0] * 12, [1] * 12, [0] * 12]
            else:
                pvc = [[1] * 6, [2] * 6, [0] * 6]

            ret = self.dh.set_pvc(ip, pvc)
            if (ret == self.result.SUCCESS):
                print("set_pvc successfully")
            else:
                print("set_pvc failed")


def main():
    Demo = DexhandDemo()
    Demo.init()
    Demo.device_info()
    Demo.set_pos()
    Demo.get_pos()
    Demo.get_ts_matrix()
    Demo.get_hand_config()
    Demo.get_pvc()
    Demo.set_pvc()

    time.sleep(1)
    Demo.reboot()

if __name__ == "__main__":
    main()

File: 《算法与竞赛操作软件对接及说明文档》/灵巧手接口文档/fourierdexhandsdk-main/example/cpp/fourier_dexhand_sdk_demo.cpp
#include <iostream>
#include <vector>
#include <iomanip>
#include <thread>

#include "dexhand.h"


int main()
{
    FdHand::DexHand dexHand;

    // init search device
    FdHand::Ret initResult = dexHand.init();
    if (initResult != FdHand::Ret::SUCCESS)
    {
        std::cerr << "init failed! error: " << static_cast<int>(initResult) << std::endl;
        return -1;
    }
    std::cout << "init successfully!" << std::endl;

    // get IP list
    std::vector<std::string> ipList = dexHand.get_ip_list();
    if (ipList.empty())
    {
        std::cerr << "no device connected" << std::endl;
        return -1;
    }

    std::cout << "connected IP list:" << std::endl;
    for (const auto &ip : ipList)
    {
        std::cout << "- " << ip << std::endl;
    }

    // create thread for each device
    std::vector<std::thread> threadList;
    for (const auto &ip : ipList)
    {
        threadList.emplace_back([&, ip]() {
            std::string deviceName = dexHand.get_name(ip);
            std::cout << "device name: " << deviceName << std::endl;

            std::string deviceType = dexHand.get_type(ip);
            std::cout << "device type: " << deviceType << std::endl;

            std::string driverVersion = dexHand.get_driver_ver(ip);
            std::cout << "driver version: " << driverVersion << std::endl;

            std::string hardwareVersion = dexHand.get_hardware_ver(ip);
            std::cout << "hardware version: " << hardwareVersion << std::endl;

            // get current position
            std::vector<float> currentPosition = dexHand.get_pos(ip);
            if (currentPosition.empty())
            {
                std::cerr << "get current position failed" << std::endl;
            }
            else
            {
                std::cout << "currentPosition: ";
                for (const auto &pos : currentPosition)
                {
                    std::cout << std::fixed << std::setprecision(2) << pos << " ";
                }
                std::cout << std::endl;
            }

            // set position
            std::vector<float> targetPosition;
            if (deviceName == "fdhv2")
            {
                targetPosition.assign(12, 1.0);
            }
            else
            {
                targetPosition.assign(6, 1.0);
            }
            FdHand::Ret setPositionResult = dexHand.set_pos(ip, targetPosition);
            if (setPositionResult == FdHand::Ret::SUCCESS)
            {
                std::cout << "set position successfully!" << std::endl;
            }
            else
            {
                std::cerr << "set position failed! error: " << static_cast<int>(setPositionResult) << std::endl;
            }

            // get ts_matrix
            if (deviceName == "fdhv2")
            {
                auto tsMatrix = dexHand.get_ts_matrix(ip);
                if (!tsMatrix.empty())
                {
                    std::cout << "tsMatrix: ";
                    for (const auto &matrix : tsMatrix)
                    {
                        for (const auto &m: matrix)
                        {
                            printf("%u ", m);
                        }
                        std::cout << std::endl;
                    }
                }
                else
                {
                    std::cerr << "get tsMatrix failed!" << std::endl;
                }
            }

            std::vector<std::vector<float>> pvc = dexHand.get_pvc(ip);
            if (pvc.empty())
            {
                std::cerr << "get pvc failed" << std::endl;
            }
            else
            {
                std::cout << "p: ";
                for (const auto &p : pvc[0])
                {
                    std::cout << std::fixed << std::setprecision(2) << p << " ";
                }
                std::cout << std::endl;

                std::cout << "v: ";
                for (const auto &v : pvc[1])
                {
                    std::cout << std::fixed << std::setprecision(2) << v << " ";
                }
                std::cout << std::endl;

                std::cout << "c: ";
                for (const auto &c : pvc[2])
                {
                    std::cout << std::fixed << std::setprecision(2) << c << " ";
                }
                std::cout << std::endl;
            }

            HandCfg_t config = dexHand.get_hand_config(ip);
            std::cout << "result " << config.result << ", ip: " << config.ip << ", gateway: " << config.gateway << ", sn: " << config.sn << std::endl;

            // reboot
            FdHand::Ret rebootResult = dexHand.reboot(ip);
            if (rebootResult == FdHand::Ret::SUCCESS)
            {
                std::cout << "reboot successfully!" << std::endl;
            }
            else
            {
                std::cerr << "reboot failed! error: " << static_cast<int>(rebootResult) << std::endl;
            }
        });
    }

    for (auto &thread : threadList)
    {
        thread.join();
    }

    return 0;
}